### ISSUE TITLE
Segmentation fault

### ISSUE THREAD

**Issue by pjrtb** [aerospike/aerospike-server#103]
Hi,

I have two crashing instances of aerospike CE (3.6.0, downgraded from 3.63 which also was crashing).

```
ldd $(which asd)
        linux-vdso.so.1 =>  (0x00007fff5cdef000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fca955d0000)
        librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007fca953c8000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fca951c4000)
        libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007fca94fab000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fca94ca5000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fca94a8f000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fca946ca000)
        /lib64/ld-linux-x86-64.so.2 (0x00007fca957ee000)
```

OS: Ubuntu 14.04.3 LTS
libc6 - 2.19-0ubuntu6.6
libgcc1 - 1:4.9.1-0ubuntu1
zlib1g - 1:1.2.8.dfsg-1ubuntu1

backtrace from gdb:
```
#0  0x000000000058fc9d in je_bitmap_set (bit=18446744073709551555, binfo=0x9fb380 <je_arena_bin_info+768>,
    bitmap=0x7fffd619d010) at include/jemalloc/internal/bitmap.h:101
#1  je_bitmap_sfu (binfo=0x9fb380 <je_arena_bin_info+768>, bitmap=0x7fffd619d010)
    at include/jemalloc/internal/bitmap.h:140
#2  arena_run_reg_alloc (bin_info=0x9fb358 <je_arena_bin_info+728>, run=0x7fffd619d000) at src/arena.c:291
#3  je_arena_tcache_fill_small (arena=0x7ffff65f0280, tbin=0x7fffc6450108, binind=7, prof_accumbytes=<optimized out>)
    at src/arena.c:1479
#4  0x00000000005a810f in je_tcache_alloc_small_hard (tcache=<optimized out>, tbin=0x7fffc6450108,
    binind=<optimized out>) at src/tcache.c:72
#5  0x0000000000585c94 in je_tcache_alloc_small (size=112, tcache=0x7fffc6450000, zero=false)
    at include/jemalloc/internal/tcache.h:303
#6  je_arena_malloc (zero=false, size=<optimized out>, arena=0x0, try_tcache=true)
    at include/jemalloc/internal/arena.h:957
#7  je_imalloct (arena=0x0, try_tcache=true, size=<optimized out>)
    at include/jemalloc/internal/jemalloc_internal.h:771
#8  je_imalloc (size=<optimized out>) at include/jemalloc/internal/jemalloc_internal.h:780
#9  malloc (size=<optimized out>) at src/jemalloc.c:929
#10 0x00000000004ff12d in cf_malloc_at (sz=105, file=0x665270 "base/thr_demarshal.c", line=591) at alloc.c:1470
#11 0x000000000048c614 in thr_demarshal (arg=<optimized out>) at base/thr_demarshal.c:591
#12 0x00007ffff7bc4182 in start_thread (arg=0x7ffd5c75c700) at pthread_create.c:312
#13 0x00007ffff6db047d in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:111
```

tail from log:
```
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:rw_dup_prole:2136) [REC NOT FOUND:1]<rw_dup_prole()>PID(193) Pstate(1):<Digest>:0xc120db8e859f5a7def2b706fae709671703eac15
Nov 09 2015 09:55:30 GMT: DEBUG (cf:jem): (jem.c:jem_set_arena:182) TID 23932 changed from arena #150 ==> #150
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:as_rw_start:1197) [PROCESS KEY] {iassd:2045} Self(bb9d6288b902500) Read(0):<Digest>:0xfdb7814a944d0ca24849c2555e1898f8d86b9495
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:internal_rw_start:807) {iassd:2045} as_write_start: duplicate partitions encountered 1 bb9abb08a902500
Nov 09 2015 09:55:30 GMT: DEBUG (demarshal): (thr_demarshal.c:thr_demarshal:472) receiving pipelined request
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:internal_rw_start:1103) write: sending request to 1 nodes
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:send_messages:366) resending rw request: {iassd:2045} node bb9abb08a902500 digest a20c4d944a81b7fd
Nov 09 2015 09:55:30 GMT: DEBUG (cf:jem): (jem.c:jem_set_arena:182) TID 23931 changed from arena #24 ==> #150
Nov 09 2015 09:55:30 GMT: DEBUG (tsvc): (thr_tsvc.c:process_transaction:542) [WRITE CASE]: Partition CK(ba4b040ae66096e)
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:as_rw_start:1197) [PROCESS KEY] {iassd:488} Self(bb9d6288b902500) Read(0):<Digest>:0xe8910facf3ff072ac4b56048e9e3274628aa291c
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:internal_rw_start:807) {iassd:488} as_write_start: duplicate partitions encountered 1 bb9abb08a902500
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:internal_rw_start:1103) write: sending request to 1 nodes
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:send_messages:366) resending rw request: {iassd:488} node bb9abb08a902500 digest 2a07fff3ac0f91e8
Nov 09 2015 09:55:30 GMT: DEBUG (cf:jem): (jem.c:jem_set_arena:182) TID 23925 changed from arena #150 ==> #150
Nov 09 2015 09:55:30 GMT: DEBUG (cf:jem): (jem.c:jem_set_arena:182) TID 23937 changed from arena #150 ==> #150
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:write_process:2851) Write Pickled: PID(3793) PState(1) Gen(63):<Digest>:0xd1fef5ffb532187028513ad9755abcd71b93fef6
Nov 09 2015 09:55:30 GMT: DEBUG (cf:jem): (jem.c:jem_set_arena:182) TID 23933 changed from arena #150 ==> #150
Nov 09 2015 09:55:30 GMT: DEBUG (cf:jem): (jem.c:jem_set_arena:182) TID 23929 changed from arena #150 ==> #24
Nov 09 2015 09:55:30 GMT: DEBUG (tsvc): (thr_tsvc.c:process_transaction:542) [WRITE CASE]: Partition CK(ba4b040ae66096e)
Nov 09 2015 09:55:30 GMT: DEBUG (cf:jem): (jem.c:jem_set_arena:182) TID 23936 changed from arena #150 ==> #150
Nov 09 2015 09:55:30 GMT: DEBUG (cf:jem): (jem.c:jem_set_arena:182) TID 23934 changed from arena #150 ==> #150
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:as_rw_start:1197) [PROCESS KEY] {iassd:3450} Self(bb9d6288b902500) Read(0):<Digest>:0x7a6db9b11841c4459d456acce542e90adb7411c0
Nov 09 2015 09:55:30 GMT: DEBUG (demarshal): (thr_demarshal.c:thr_demarshal:472) receiving pipelined request
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:internal_rw_start:807) {iassd:3450} as_write_start: duplicate partitions encountered 1 bb9abb08a902500
Nov 09 2015 09:55:30 GMT: DEBUG (tsvc): (thr_tsvc.c:process_transaction:542) [WRITE CASE]: Partition CK(ba4b040ae66096e)
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:internal_rw_start:1103) write: sending request to 1 nodes
Nov 09 2015 09:55:30 GMT: DEBUG (tsvc): (thr_tsvc.c:process_transaction:542) [WRITE CASE]: Partition CK(ba4b040ae66096e)
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:send_messages:366) resending rw request: {iassd:3450} node bb9abb08a902500 digest 45c44118b1b96d7a
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:as_rw_start:1197) [PROCESS KEY] {iassd:921} Self(bb9d6288b902500) Read(0):<Digest>:0x99a303476d8fdf374b8657b8c9c93fc28cbbfa7b
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:rw_process_ack:1911) {iassd:586} rw_process_ack: Processing response 377401554d18024a WRITE
Nov 09 2015 09:55:30 GMT: DEBUG (cf:jem): (jem.c:jem_set_arena:182) TID 23931 changed from arena #150 ==> #150
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:internal_rw_start:807) {iassd:921} as_write_start: duplicate partitions encountered 1 bb9abb08a902500
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:as_rw_start:1197) [PROCESS KEY] {iassd:501} Self(bb9d6288b902500) Read(0):<Digest>:0xf521b2d962948be061e553698318ccb4bf6c735f
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:internal_rw_start:1103) write: sending request to 1 nodes
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:internal_rw_start:807) {iassd:501} as_write_start: duplicate partitions encountered 1 bb9abb08a902500
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:send_messages:366) resending rw request: {iassd:921} node bb9abb08a902500 digest 37df8f6d4703a399
Nov 09 2015 09:55:30 GMT: DEBUG (cf:jem): (jem.c:jem_set_arena:182) TID 23932 changed from arena #150 ==> #150
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:internal_rw_start:1103) write: sending request to 1 nodes
Nov 09 2015 09:55:30 GMT: DEBUG (cf:jem): (jem.c:jem_set_arena:182) TID 23931 changed from arena #150 ==> #150
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:send_messages:366) resending rw request: {iassd:501} node bb9abb08a902500 digest e08b9462d9b221f5
Nov 09 2015 09:55:30 GMT: DEBUG (tsvc): (thr_tsvc.c:process_transaction:542) [WRITE CASE]: Partition CK(ba4b040ae66096e)
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:write_process:2851) Write Pickled: PID(832) PState(1) Gen(279):<Digest>:0x404362dd40db2e5c86bb442f23db18157a9e822d
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:rw_process_ack:1911) {iassd:2045} rw_process_ack: Processing response a20c4d944a81b7fd WRITE
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:as_rw_start:1197) [PROCESS KEY] {iassd:1651} Self(bb9d6288b902500) Read(0):<Digest>:0x73263891a6e839bf38e5c7a91fd6dfc4b430a733
Nov 09 2015 09:55:30 GMT: DEBUG (record): (record.c:as_partition_vinfoset_unpickle:1764) received vinfo set with 0 size, unusual RW
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:finish_rw_process_dup_ack:1627) finish_rw_process_dup_ack: received message with record properties
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:internal_rw_start:807) {iassd:1651} as_write_start: duplicate partitions encountered 1 bb9abb08a902500
Nov 09 2015 09:55:30 GMT: DEBUG (record): (record.c:as_record_flatten:1215) flatten start: 
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:internal_rw_start:1103) write: sending request to 1 nodes
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:send_messages:366) resending rw request: {iassd:1651} node bb9abb08a902500 digest bf39e8a691382673
Nov 09 2015 09:55:30 GMT: DEBUG (record): (record.c:as_partition_vinfoset_unpickle:1764) received vinfo set with 0 size, unusual RW
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:write_process:2851) Write Pickled: PID(2006) PState(1) Gen(3):<Digest>:0xd6376836cda8c43b02014a510396fcf23e4c711a
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:internal_rw_start:1103) write: sending request to 1 nodes
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:finish_rw_process_dup_ack:1627) finish_rw_process_dup_ack: received message with record properties
Nov 09 2015 09:55:30 GMT: DEBUG (rw): (thr_rw.c:send_messages:366) resending rw request: {iassd:2045} node bb9abb08a902500 digest a20c4d944a81b7fd
```

**Comment by pjrtb** [comment]
Also, there is 
```
write-block-size 1M
```

in the configuration.

**Comment by pjrtb** [comment]
Stack trace from logs on the second machine:

```
Nov 09 2015 18:23:49 GMT: WARNING (as): (signal.c::163) stacktrace: frame 0: /usr/bin/asd(as_sig_handle_segv+0x5d) [0x473606]
Nov 09 2015 18:23:49 GMT: WARNING (as): (signal.c::163) stacktrace: frame 1: /lib/x86_64-linux-gnu/libc.so.6(+0x36d40) [0x7f45000b9d40]
Nov 09 2015 18:23:49 GMT: WARNING (as): (signal.c::163) stacktrace: frame 2: /usr/bin/asd() [0x58763a]
Nov 09 2015 18:23:49 GMT: WARNING (as): (signal.c::163) stacktrace: frame 3: /usr/bin/asd() [0x58a51b]
Nov 09 2015 18:23:49 GMT: WARNING (as): (signal.c::163) stacktrace: frame 4: /usr/bin/asd(cf_free_at+0x7e) [0x4f81d8]
Nov 09 2015 18:23:49 GMT: WARNING (as): (signal.c::163) stacktrace: frame 5: /usr/bin/asd(msg_reset+0x4c) [0x4ffb76]
Nov 09 2015 18:23:49 GMT: WARNING (as): (signal.c::163) stacktrace: frame 6: /usr/bin/asd(as_fabric_msg_put+0x2f) [0x4cd0be]
Nov 09 2015 18:23:49 GMT: WARNING (as): (signal.c::163) stacktrace: frame 7: /usr/bin/asd(write_request_destructor+0x43) [0x4a788f]
Nov 09 2015 18:23:49 GMT: WARNING (as): (signal.c::163) stacktrace: frame 8: /usr/bin/asd(rw_process_ack+0x29a) [0x4b4d41]
Nov 09 2015 18:23:49 GMT: WARNING (as): (signal.c::163) stacktrace: frame 9: /usr/bin/asd(write_msg_fn+0x2df) [0x4b70c4]
Nov 09 2015 18:23:49 GMT: WARNING (as): (signal.c::163) stacktrace: frame 10: /usr/bin/asd(fabric_process_read_msg+0x178) [0x4ce876]
Nov 09 2015 18:23:49 GMT: WARNING (as): (signal.c::163) stacktrace: frame 11: /usr/bin/asd(fabric_process_readable+0x3d) [0x4cee9a]
Nov 09 2015 18:23:49 GMT: WARNING (as): (signal.c::163) stacktrace: frame 12: /usr/bin/asd(fabric_worker_fn+0x403) [0x4cf33e]
Nov 09 2015 18:23:49 GMT: WARNING (as): (signal.c::163) stacktrace: frame 13: /lib/x86_64-linux-gnu/libpthread.so.0(+0x8182) [0x7f4500f91182]
Nov 09 2015 18:23:49 GMT: WARNING (as): (signal.c::163) stacktrace: frame 14: /lib/x86_64-linux-gnu/libc.so.6(clone+0x6d) [0x7f450017d47d]
```

**Comment by wchu-citrusleaf** [comment]
Do you have batch requests to the system? In 3.6.0-3.6.3 there are some batch related memory stomps.

**Comment by pjrtb** [comment]
We have batch gets.

**Comment by xiaosuo** [comment]
I have the same problem, and I have batch requests. At the same time, the same issue happens with 3.6.3.

**Comment by MichaelDiep** [comment]
We just released a new Aerospike version 3.6.4.  Please upgrade to 3.6.4, this version fixes current batch issue.
